kernel/memory/src/lib.rs

```diff
@@ -220,7 +221,15 @@ pub fn init(
     debug!("Initialized new frame allocator!");
     frame_allocator::dump_frame_allocator_state();
 
-    page_allocator::init(VirtualAddress::new_canonical(boot_info.kernel_end()?.value()))?;
+    page_allocator::init(
+        VirtualAddress::new(
+            Mapper::from_current()
+                .translate(boot_info.kernel_end()?)
```

Comment:
is this right? Looking at the calculation of `kernel_end`, it seems like an exclusive upper bound. I didn't dig into it in detail, but what happens if `kernel_end()` is aligned to the start of the next page, causing this `translate()` action to fail?

Reply:
ok on second inspection, this is a bit odd. The boot_info provides the kernel end vaddr, which we then translate to physical, which we the use to obtain the identity vaddr? I legit cannot tell if that's equivalent to the previous code, lol